lcls_tools/data_analysis/archiver.py

```diff
@@ -18,82 +20,105 @@
 TIMEOUT = 3
 
 
+# Adding a data class for easier data handling (so that external functions can
+# invoke .values and .timestamps instead of having to remember dictionary keys
+@dataclass()
+class ArchiverData:
+    # todo make this not a union
+    # Currently one list if timestamps are shared by all PVs else a dict of
+    # pv -> timstamps
+    timeStamps: Union[List[datetime], Dict[str, List[datetime]]]
+    values: Dict[str, List]
+
+
 class Archiver(object):
+
+    # machine is a string that is either "lcls" or "facet"
     def __init__(self, machine):
         # type: (str) -> None
         self.url_formatter = ARCHIVER_URL_FORMATTER.format(MACHINE=machine)
 
     def getDataAtTime(self, pvList, timeRequested):
```

Comment:
```suggestion
    def get_data_at_time(self, pv_list: List[str], time_requested: datetime):
```
Depending on what python version you're using.  Main take away is use snake case for all things python.

Anyway, bouncing out here, but I'd love to help you finish this.  

Rules:
Never leave a PR open for more than 2 weeks... even at a national lab
Quick iteration is best
Write tests
No magic numbers and no hard coding
If you go in on type hinting/validation, go all in

Reply:
I forgot to say that I hope all is well at SLAC.  I'm super happy to see some great people making great things happen!